db/database.go

```diff
@@ -2567,3 +2568,31 @@ func (db *Database) DataStoreNames() base.ScopeAndCollectionNames {
 	}
 	return names
 }
+
+// GetCollectionIDs will return all collection IDs for all collections configured on the database
+func (db *DatabaseContext) GetCollectionIDs() []uint32 {
+	collIDs := make([]uint32, 0)
+	for id, _ := range db.CollectionByID {
+		collIDs = append(collIDs, id)
+	}
+	return collIDs
+}
+
+// PurgeDCPMetadata will purge all DCP metadata from previous run in the bucket, used to reset dcp client to 0
+func PurgeDCPMetadata(ctx context.Context, database *DatabaseContext, metadataKeyPrefix string, taskID string) error {
```

Comment:
```suggestion
// PurgeDCPCheckpoints will purge all DCP checkpoints.
func PurgeDCPCheckpoints(ctx context.Context, database *DatabaseContext, checkpointPrefix string) error {
```

This function is now more generic than background tasks, so I think we should move `base.InfofCtx(ctx, base.KeyDCP, "Purging invalid checkpoints for background task run %s", taskID)` outside. I also think we should jsut checkpointPrefix since otherwise this reads more like `base.MetadataKeys`.